hadml/models/components/encoder_transformer.py

```diff
@@ -0,0 +1,67 @@
+import torch, os, pickle
+
+
+class Generator(torch.nn.Module):
+    """ Generator implemented as a encoder-only transformer model """
+
+    def __init__(
+        self,
+        noise_dim=4,            # Arbitrary number (noise dimensionality)
+        cluster_kins_dim=8,     # Cluster four-momentum, two quark types, phi, theta
+        n_quarks=2,             # Number of quark types in cluster_kins_dim
+        quark_types=16,         # Quark PIDs: (+/-) 1-8 -> transformation -> 0-16
+        hadron_kins_dim=4,      # Hadron four-momentum
+        embedding_dim=32,       # Arbitrary number (but the same for the discriminator)
+        quark_embedding_dim=4,  # Arbitrary number (quark embedding dimensionality)
+        n_heads=8,              # Encoder architecture hyperparameter
+        pid_map_filepath=None   # For getting information about the number of hadron most common IDs
+    ):
+        super().__init__()
+        self.hadron_kins_dim = hadron_kins_dim
+        with open(os.path.normpath(pid_map_filepath), "rb") as f:
+            n_hadron_types = len(pickle.load(f)) + 1
+        self.quark_type_embedding_layer = torch.nn.Embedding(quark_types, quark_embedding_dim)
+        self.input_embedding_layer = torch.nn.Linear(
+            noise_dim + cluster_kins_dim - n_quarks + n_quarks * quark_embedding_dim, embedding_dim
```

Comment:
Removing quark types from kinematics would simplify this line a bit. Assuming that removing it makes sense.